src/core/text/TextMetrics.js

```diff
@@ -103,86 +108,139 @@ export default class TextMetrics
     {
         const context = canvas.getContext('2d');
 
-        // Greedy wrapping algorithm that will wrap words as the line grows longer
-        // than its horizontal bounds.
-        let result = '';
-        const firstChar = text.charAt(0);
-        const lines = text.split('\n');
-        const wordWrapWidth = style.wordWrapWidth;
-        const characterCache = {};
-
-        for (let i = 0; i < lines.length; i++)
+        let line = '';
+        let width = 0;
+        let lines = '';
+        const cache = {};
+
+        // ideally there is letterSpacing after every char except the last one
+        // t_h_i_s_' '_i_s_' '_a_n_' '_e_x_a_m_p_l_e_' '_!
+        // so for convenience the above needs to be compared to width + 1 extra space
+        // t_h_i_s_' '_i_s_' '_a_n_' '_e_x_a_m_p_l_e_' '_!_
+        // ________________________________________________
+        // And then the final space is simply no appended to each line
+        const wordWrapWidth = style.wordWrapWidth + style.letterSpacing;
+
+        // store the width of calculated character(s)
+        function getFromCache(key)
```

Comment:
I have a personal preference for having these functions stored outside of this static wordWrap function, so that they are not re-created every time the wordWrap function is called